### ISSUE TITLE
SerializerCodeGenerationConfiguration.IsRecursive doesn't track generic type

### ISSUE THREAD

**Issue by neuecc** [msgpack/msgpack-cli#106]
If recursive type in generic(array, list, dictionary, etc...), 
SerializerCodeGenerator doesn't generate generic element type.

```csharp
var settings = new SerializerCodeGenerationConfiguration
{
    IsRecursive = true,
};
var codes = SerializerGenerator.GenerateCode(settings, typeof(A_Class));

foreach (var item in codes)
{
    // only A_ClassSerializer but my expected is generates A_ClassSerializer and B_ClassSerializer
    Console.WriteLine(item);
}
```

```csharp
public class A_Class
{
    public B_Class[] Prop_B { get; set; }
}

public class B_Class
{
    public int MyProperty { get; set; }
}
```

**Comment by yfakariya** [comment]
Thank you for reporting.
I agree the element type and generic arguments should also be targeted.

**Comment by yfakariya** [comment]
Fixed in 0.6.1.
